### PR TITLE
Restructure/reorganize content in SM introduction pages 

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description

Resolves [#3235](https://github.com/camunda/camunda-docs/issues/3235), relates to [#199](https://github.com/camunda/developer-experience/issues/199). 

This PR separates out the consolidation and clean-up of the intro, architecture, and installation overview pages from any content additions or adjustments to other pages. The changes are currently confined to `/next`, but can/will be backported + rerouted after an initial review, if needed.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [inline docs/self-managed/about-self-managed.md:14]: Leaving an expansion on content here for #1055 - just a quick rewrite for now.

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [inline docs/self-managed/about-self-managed.md:23]: I believe these architecture diagrams are a little out of date, so I attempted to include that these were example/sample deployments, only, until they can be updated.

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [inline docs/self-managed/about-self-managed.md:30]: The "additional considerations" section in the original architecture doc includes this:
> For components with Anti-Affinity enabled, like Zeebe Broker, the Kubernetes Nodes should be equal to or greater than the number of the Pod replicas. Otherwise, some of its Pods will not be scheduled and will be in a "Pending" state.
As far as I could tell, this is the only place we talk about which (if any) components have anti-affinity enabled, or mention what to do in that case/"Pending" state pods? I don't think this diagram section is the right place for this information, even in passing, and if this is something SM users will need to know, it should be called out more directly in appropriate sections...? I can spin off a new issue to research this if this line of thought seems correct.

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [inline docs/self-managed/about-self-managed.md:25]: I think this list is more appropriate on the install page, where we have to talk about how components can be installed in different configurations anyway, but could also see the argument for it being here instead (the goal being to only list it in one place instead of duplicating it). Ideally the About page would include more info about what & why to justify this content move.

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [inline docs/self-managed/about-self-managed.md:10]: I discussed the use of the Camunda 7 references with @akeller a few weeks ago... In the effort of reducing the amount of C7 content in the SM docs (and potential confusion), this seems like an easy way to start.

File: docs/self-managed/about-self-managed.md
Comment (akeller) [reply docs/self-managed/about-self-managed.md:30]: I added @theburi as a reviewer to the overall PR, but I'm hoping he could shed some light on this specific topic. Is this something we can address here quickly @theburi, or should @conceptualshark create a new issue and discuss it with you separately?

File: docs/self-managed/about-self-managed.md
Comment (akeller) [reply docs/self-managed/about-self-managed.md:25]: If we put this info in a reusable component, we could duplicate it without the maintenance overhead. Would that change your thoughts here?

**Review by akeller — COMMENTED** [review]
In general, I like the direction these changes are going and my hope is the feedback that does come in isn't significantly blocking or can be addressed in iterations (for example, diagrams may be stale).

As I read through this, I'm wondering if "Ingress" is clear to users. Can you add this to the [glossary](https://docs.camunda.io/docs/next/reference/glossary/)? In this PR Ingress is used by itself or "Ingress objects," and I'm not sure they need to be different. 

Lastly, don't get caught off guard when you backport these changes and need a redirect to remove the Architecture page. We don't do the same checks against `/next/` content.

**Comment by akeller** [comment]
I'm adding @hisImminence as a review in case he wants to involve other engineers in the Self-Managed space in reviewing these changes.

**Review by theburi — DISMISSED** [review]

File: docs/self-managed/setup/overview.md
Comment (hisImminence) [inline docs/self-managed/setup/overview.md:39]: One thought here: would it make sense to have the Docker Compose part as a separate bullet point (would make if fmpov clearer that its not related to the docker setup), and only local!

File: docs/self-managed/about-self-managed.md
Comment (hisImminence) [reply docs/self-managed/about-self-managed.md:42]: I would argue that 'it is possible to deploy HA mode', and therefore the statement is correct :P I feel it would sound more confusing if we state its not HA or partially.

Would it be an idea to create a follow up to create a guide or make the HCs configuration more easy for customers to set up all apps in HA mode then?

**Comment by hisImminence** [comment]
> I'm adding @hisImminence as a review in case he wants to involve other engineers in the Self-Managed space in reviewing these changes.

Thanks @akeller, reviewed and besides the 2 commands I left lgtm!

**Review by hisImminence — DISMISSED** [review]

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [reply docs/self-managed/about-self-managed.md:30]: I'm going to go ahead and remove the notes from this section of the docs, which is meant to be more of an overview, and created an issue for discussion/creation of a better place for these recommendations/constraints/etc to live (https://github.com/camunda/camunda-docs/issues/4075).

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [reply docs/self-managed/about-self-managed.md:42]: @hisImminence @theburi Do you think this diagram with a theoretical HA mode is adding anything for a technical audience? I could remove it if it's only adding to confusion, and make an issue to create the follow-up HA guide. Or, leave it for now, and make that guide to link to from here when it's done.

**Comment by conceptualshark** [comment]
@akeller I broke off a separate issue to add Ingress to the glossary [here](https://github.com/camunda/camunda-docs/pull/4077), and will follow up with some changes to this PR.

File: docs/self-managed/about-self-managed.md
Comment (hisImminence) [reply docs/self-managed/about-self-managed.md:42]: >  I could remove it if it's only adding to confusion, and make an issue to create the follow-up HA guide

You are referring to the diagram on [this page](https://docs.camunda.io/docs/self-managed/setup/install/), or? 

I would go with this option: I could remove it if it's only adding to confusion, and make an issue to create the follow-up HA guide.

Comment (conceptualshark) [reply docs/self-managed/about-self-managed.md:42]: Specifically the high availability diagram [here](https://docs.camunda.io/docs/self-managed/platform-architecture/overview/#deployment). I've gone ahead and removed it in this iteration, and set up [this issue](https://github.com/camunda/camunda-docs/issues/4080) to look into it outside of this cleanup PR.

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [reply docs/self-managed/about-self-managed.md:25]: I've tried this iteration with a reused partial, which I think works nicely in both pages. If they start to diverge in content we can separate them out again.

File: docs/self-managed/setup/overview.md
Comment (conceptualshark) [reply docs/self-managed/setup/overview.md]: I pulled out docker compose in the latest push, and put it in its own mention, to keep this only a list of option for production-viable deployments.

**Comment by akeller** [comment]
Lots of changes, so I added the `deploy` label for reviewers 😎

**Review by akeller — COMMENTED** [review]
I made one recommendation, but IMO, you are good to backport.

**Comment by akeller** [comment]
@conceptualshark how is this going? Are you ready for a review? Looking for feedback from specific people? How can I help? 😄

**Comment by conceptualshark** [comment]
@akeller I promise this isn't as large as it seems, but backporting was a bit of a mess! I'd love a final sign-off on the redirects and a sanity check on the backported links. I think it's in a good state now.

File: docs/self-managed/about-self-managed.md
Comment (theburi) [inline docs/self-managed/about-self-managed.md:38]: We refer to these as Components. Web Applications is not a topology term that we use.  It also does not represent them fully

Comment (conceptualshark) [reply docs/self-managed/about-self-managed.md:38]: I think the usage here shouldn't capture each component fully, only represent the web interface, and should explicitly separate them out from Zeebe in terms of access.

There's a number of places where we use web app/application like this, so if it shouldn't be used at all, let me know if I should create a separate ticket to update it elsewhere.

Comment (akeller) [reply docs/self-managed/about-self-managed.md:38]: I could not craft a good definition of when to use components vs. web applications off the top of my head. If I had to reason out loud, it would be the following: Modeler is the component, but Web Modeler is a web application. I don't think Identity is a web application, but I don't know how I would verify that.

For internal resources, I would recommend this [PMM-maintained page](https://confluence.camunda.com/display/HAN/Internal+Guidance+on+Product+Name%2C+Version%2C+and+Packaging). You can also follow up with @karl-heinsen since I hit a dead end [here](https://camunda.slack.com/archives/C0589TC88UU/p1698325079905339).

File: docs/self-managed/react-components/components.md
Comment (akeller) [reply docs/self-managed/react-components/components.md:11]: This works but might have some weird side effects. I think you want the following:

```suggestion
- Console [<span class="badge badge--enterprise-only">Enterprise only</span>](/reference/licenses.md#console)
```

File: docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md
Comment (akeller) [inline docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md:28]: I got into a habit of using the docs area instead of trying to walk the file structure with `../`'s. Can you bring this to a team meeting so we can discuss it? No action required here.

**Review by akeller — CHANGES_REQUESTED** [review]
@conceptualshark have a look at the link vale is complaining about. Linking to `/docs/` will not respect the version drop down. 

My other feedback is not blocking for this PR.

File: docs/self-managed/about-self-managed.md
Comment (conceptualshark) [reply docs/self-managed/about-self-managed.md:38]: Would "web components" be preferable? The entirety of this section is focused on separating out the different points of access, so "Components" isn't inclusive enough as Zeebe isn't included, and we're also not referring to the entirety of the component - only the method of access. Looking for any term that is inclusive of all components with a served log-in page, and excludes those without.

```suggestion
In this configuration, Camunda 8 Self-Managed can be accessed as follows:
- Web components: `https://camunda.example.com/[identity|operate|optimize|tasklist|modeler]`
  - Web Modeler also exposes a WebSocket endpoint on `https://camunda.example.com/modeler-ws`. This is only used by the application itself and not supposed to be accessed by users directly.
- Keycloak authentication: `https://camunda.example.com/auth`
- Zeebe gateway: `grpc://zeebe.camunda.example.com`
```

Comment (akeller) [reply docs/self-managed/about-self-managed.md:38]: I would back either "web applications" or "web components" to get this out the door.

**Review by akeller — APPROVED** [review]
🔍 Great detail-oriented work here, particularly with the tricky enterprise badges! I know there are some opens still, but I'm giving this my pre-approval!

**Review by akeller — APPROVED** [review]
Pre-approving! I know you still had an open item, but this gets the 🟢 from me.

File: docs/self-managed/about-self-managed.md
Comment (theburi) [reply docs/self-managed/about-self-managed.md:38]: Web Modeller has three parts - NodeJS server that exposes web interface, REST API server and Socket server. Identity has also web interface.  'Web Component' is to restrictive and diminishes the value of our product. We typically refer to zeebe, optimize, operate as Component or C8.
Can we use 'Component' going forward?

**Review by theburi — DISMISSED** [review]

**Comment by conceptualshark** [comment]
Called out the components explicitly to remove any reference to "web" in either components or applications; carved out https://github.com/camunda/developer-experience/issues/370 to tackle the use of these terms throughout the docs.

**Comment by conceptualshark** [comment]
Waiting on the results of https://github.com/camunda/camunda-docs/pull/3888 before merging to more easily incorporate those changes (terms approved by support and legal, etc) - should be within 24 hours.

**Comment by conceptualshark** [comment]
@akeller Got the okay to finish this ahead of delays/necessary refactoring of #3888! 🙏 Just needs the final stamp of approval.

**Pull request merged**
